internal/setup/job.go

```diff
@@ -249,3 +244,23 @@ func newHelmGeneratorConfig(task *optimizev1beta2.SetupTask) *helmGeneratorConfi
 
 	return cfg
 }
+
+func labels(expName string, trialName string, tasks []optimizev1beta2.SetupTask) map[string]string {
+	labels := map[string]string{
+		optimizev1beta2.LabelExperiment: expName,
+		optimizev1beta2.LabelTrial:      trialName,
+		optimizev1beta2.LabelTrialRole:  "trialSetup",
+	}
+
+	for _, task := range tasks {
+		for k, v := range task.Labels {
+			if _, ok := labels[k]; ok {
+				continue
+			}
+
+			labels[k] = v
+		}
+	}
+
+	return labels
```

Comment:
I had to double take at the double loop, `ok ... continue` bit, maybe instead:
```
labels := make(map[string]string, len(task.Labels)+3)
for k, v := range task.Labels {
  labels[k] = v
}
labels[optimizev1beta2.LabelExperiment] = expName
labels[optimizev1beta2.LabelTrial] = trialName
labels[optimizev1beta2.LabelTrialRole] = "trialSetup"
return labels
```

Reply:
Went back and forth with which way to go on that. Fine with changing it to the above.